tests/unit_tests/test_model.py

```diff
@@ -32,3 +282,257 @@ def test_import_properties(run_in_tmpdir, mpi_intracomm):
     cell = model_with_properties.geometry.get_all_cells()[1]
     assert cell.temperature == [600.0]
     assert cell.fill.get_mass_density() == pytest.approx(5.0)
+
+
+def test_run(run_in_tmpdir, pin_model_attributes, mpi_intracomm):
+    mats, geom, settings, tals, plots, _, _ = pin_model_attributes
+    test_model = openmc.Model(geom, mats, settings, tals, plots)
+
+    # This case will run by getting the k-eff and tallies for command-line and
+    # C API execution modes and ensuring they give the same result.
+    sp_path = test_model.run(output=False)
+    with openmc.StatePoint(sp_path) as sp:
+        cli_keff = sp.k_combined
+        cli_flux = sp.get_tally(id=1).get_values(scores=['flux'])[0, 0, 0]
+        cli_fiss = sp.get_tally(id=1).get_values(scores=['fission'])[0, 0, 0]
+
+    test_model.init_lib(output=False, intracomm=mpi_intracomm)
+    sp_path = test_model.run(output=False)
+    with openmc.StatePoint(sp_path) as sp:
+        lib_keff = sp.k_combined
+        lib_flux = sp.get_tally(id=1).get_values(scores=['flux'])[0, 0, 0]
+        lib_fiss = sp.get_tally(id=1).get_values(scores=['fission'])[0, 0, 0]
+
+    # and lets compare results
+    assert abs(lib_keff - cli_keff) < 1e-13
+    assert abs(lib_flux - cli_flux) < 1e-13
+    assert abs(lib_fiss - cli_fiss) < 1e-13
```

Comment:
In all these cases where you're checking if two floating point values are really close, I'd recommend using:
```python
from pytest import approx

assert x == approx(y)
```
It looks like you have a bunch of cases down below that also use the same pattern and could be updated.

For the comparison of the keff values, you'll need to compare the nominal values (`lib_keff.n == approx(cli_keff.n)`)